### ISSUE TITLE
Handling illegal values in group codes

### ISSUE THREAD

**Issue by iceblu3710** [gdsestimating/dxf-parser#19]
I have noticed a few times now that programs are outputting illegal values with group codes. A simple example is

```
POLYLINE
  5
497
  8
0
 66
     1
 10
0.0
 20
0.0
 30
0.0
 70
   129
  0
VERTEX
  5
57E
  8
0
 10
210.0
 20
195.0
 30
0.0
  0
VERTEX
  5
57F
  8
0
 10
10.0
 20
195.0
 30
0.0
  0
VERTEX
  5
580
  8
0
 10
10.0
 20
145.0
 30
0.0
  0
VERTEX
  5
581
  8
0
 10
210.0
 20
145.0
 30
0.0
  0
SEQEND
  5
674
  8
0
  0
```

So this closed shape is listed as false because `70: 129` is illegal, default to `70: 0`. The program that output simpily did a +1 to close the shape. 

DraftSight reads in the file properly however so there must be some error handling. Possibly an `if val > 128 then val -=128` ??

**Comment by bzuillsmith** [comment]
actually, 129 is probably valid. It looks like we are not handling that value properly in the parser. It is a boolean flag value so 129 would be both shape and plinegen are true. 1 means only shape is true, and plinegen is false. It should be a simple fix.

**Comment by iceblu3710** [comment]
Ahh I didn't think of combined values. So the exporter is definatly just doing a +1. That also explains values like 70:2

I am out and about all day but if you havent fixed it yet I will put in a PR.

**Comment by iceblu3710** [comment]
Weird behavior going on here. 

```
entity.shape = (curr.value === 1);
```

That code works great when evaluated individually. BUT, some files `entity.shape` remains `false` even though the statement is true?!

Attached is an example of such behaviour.
[shapesIssue-polylines_r12.dxf.txt](https://github.com/gdsestimating/dxf-parser/files/1064638/shapesIssue-polylines_r12.dxf.txt)

**Comment by iceblu3710** [comment]
Hmm, I was reading the code in the source folder and it IS `entity.shape = (curr.value & 1 === 1)` but after browserfying it becomes `entity.shape = (curr.value === 1);`  

I will try to update the libs in the project and see if it's being optimized away or something.

**Comment by iceblu3710** [comment]
https://github.com/gdsestimating/dxf-parser/blob/v0.4.6/lib/DxfParser.js#L1074

**Comment by bzuillsmith** [comment]
Ahhh, I must not have published it since it got fixed. I will try to do
that soon.

**Comment by iceblu3710** [comment]
Are you going to push a minor version for npm includes?

**Comment by bzuillsmith** [comment]
Published in 4.7. Thanks @iceblu3710
